### PR TITLE
[farmhash] support Windows build

### PR DESCRIPTION
**Pull request by luncliff**
### What does your PR fix?  

Fix makefile build failures in Windows.

### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  

Removed `farmhash` annotations in `ci.baseline.txt`.

* `x64-windows`
* `x86-windows`
* `x64-uwp`
* `x64-windows-static`

### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  

Yep. All kind of feedback will be welcomed :D

### CODE REVIEW

**Review by ras0219-msft — COMMENTED** [review]
Thanks for the PR!

File: ports/farmhash/fix-windows.patch [hunk 1]
```diff
@@ -0,0 +1,343 @@
+diff --git a/src/farmhash.cc b/src/farmhash.cc
```
[the remark this answers is not in the document]
Comment (luncliff) [reply ports/farmhash/fix-windows.patch:1]: This looks much better. I will make a PR then apply this.

**Comment by JonLiu1993** [comment]
@luncliff ,Could you please take a look?
````
In file included from .././../src/97573ad06d-b8263bbc35.clean/src/farm-test.cc:36:
.././../src/97573ad06d-b8263bbc35.clean/src/farmhash.cc:11846:44: error: missing binary operator before token "("
11846 | #if defined(__has_builtin) && __has_builtin(__builtin_unreachable)
      |                                            ^
make[2]: *** [Makefile:624: farm-test.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [Makefile:473: all-recursive] Error 1
make: *** [Makefile:384: all] Error 2
````

**Comment by luncliff** [comment]
@JonLiu1993 Yeah. I'm checking with my laptop now. I will update the part when I find a good answer.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**